### DISCUSSION TITLE
Main process debug setup for PHPStorm

### DISCUSSION THREAD

**Discussion by www-m-www** [cawa-93/vite-electron-builder#1016] [Q&A]
**Describe the bug**
I'm trying to debug _Main_ process in **PHPStorm**, but it I cannot make it work. The **VSCode** setup (_.vscode/launch.json_) seems to work fine for **VSCode**, but I'd rather use **PHPStorm** for the whole project, instead of switching to **VSCode** just to debug the main process. Can you please provide clear instruction how to setup **PHPStorm** to debug the main process. Thank you.

**Comment by sama12386** [comment]
For full debugging in any Jet Brains IDE (WebStorm, IntelliJ, etc.):

In `scripts/watch.js:48`, add these arguments to the electron launch:
```
electronApp = spawn(String(electronPath), [".", "--sourcemap","--remote-debugging-port=XXXX"], {
```

Then create an npm run configuration. 
Command: **run**
Scripts: **watch**
(All else default)

To monitor the renderer process, create an Attach to Node.js/Chrome run configuration.
Host: **localhost**
Port: **[XXXX]**
All else default

You can then create a Compound run configuration and run both of those configs at the same time.
